W-14975729-SFTP-2.4-Document-Identity-file-issue-KS (#2749)
mulesoft/docs-connectors: sftp/2.4/modules/ROOT/pages/sftp-connector-troubleshooting.adoc

= Troubleshooting SFTP Connector - Mule 4

> To troubleshoot Anypoint Connector for SFTP (SFTP Connector), become familiar with the information about troubleshooting connectivity, permissions and server-side issues, troubleshooting Mule application issues, interpreting commonly thrown messages, and troubleshooting identity file format issues.

Reviewer: ```suggestion
To troubleshoot Anypoint Connector for SFTP (SFTP Connector), become familiar with the information about troubleshooting connectivity, permissions, and server-side issues, troubleshooting Mule application issues, interpreting commonly thrown messages, and troubleshooting identity file format issues.
```

  The file to write was deleted.

> == Troubleshoot SFTP Connector Issues with Identity File in .ppk Format

Reviewer: ```suggestion
== Troubleshoot SFTP Connector Issues with a PPK Identity File 
```

> When using the SFTP Connector version 2.0.0 and later with the Preferred Authentication Method set to `PUBLIC_KEY`, you might face issues if the Identity File is in `.ppk` format. This is due to a limitation with the `Apache Mina library`, which does not support the `.ppk` file format, causing the application to fail.

Reviewer: ```suggestion
When using the SFTP Connector version 2.0.0 and later with the preferred authentication method set to `PUBLIC_KEY`, you can face issues if the `Identity` file is PPK. The `Apache Mina library` doesn’t support the PPK file format, so the application fails.
```

Reviewer: Look at the file types and file extensions, and font choices in our style guide:
https://confluence.internal.salesforce.com/display/MTDT/MuleSoft+CX+Writing+Style+Reference#:~:text=file%20types%20and%20file%20extensions

Author: Thank you @Crispy-Salesforce. I wasn’t aware of this part.

To resolve this issue:

> . Convert the Identity File format from `.ppk` to `.pem` using PuTTYgen.

Reviewer: ```suggestion
. Convert the PPK `Identity` file to PEM by using `PuTTYgen`.
```

> . Update the configuration file to use the `.pem` file.

Reviewer: ```suggestion
. Update the configuration file to use the PEM file.
```
